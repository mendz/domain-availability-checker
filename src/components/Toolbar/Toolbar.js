import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import Button from '../UI/Button/Button';
import ButtonInfo from '../UniqueButtons/ButtonInfo/ButtonInfo';
import Modal from '../UI/Modal/Modal';
import Info from '../Info/Info';
import AuthModal from '../UniqueButtons/AuthModal';

import classes from './Toolbar.module.css';

class Toolbar extends Component {
  state = {
    showInfo: false,
  };

  showInfoModal = () => {
    this.setState({ showInfo: true });
  };

  closeInfoModal = () => {
    this.setState({ showInfo: false });
  };

  goToHistory = () => {
    this.props.history.push('/history');
  };

  goBack = () => {
    this.props.history.push('/');
  };

  render() {
    const { pathname } = this.props.location;
    let toolbar = null;
    if (pathname !== '/') {
      toolbar = <Button onClick={this.goBack}>Back Home</Button>;
    } else {
      toolbar = (
        <>
          <Button
            onClick={this.goToHistory}
            name="go-to-history"
            disabled={this.props.isChecking}
          >
            History
          </Button>
          <div className={classes.LeftButtons}>
            <AuthModal disabled={this.props.isChecking} />
            <ButtonInfo
              clicked={this.showInfoModal}
              disabled={this.props.isChecking}
            />
            <Modal show={this.state.showInfo} closed={this.closeInfoModal}>
              <Info />
            </Modal>
          </div>
        </>
      );
    }

    return <div className={classes.Toolbar}>{toolbar}</div>;
  }
}

const mapStateToProps = state => ({
  isChecking: state.domainList.checking,
});

export default connect(mapStateToProps)(withRouter(Toolbar));

export { Toolbar };
