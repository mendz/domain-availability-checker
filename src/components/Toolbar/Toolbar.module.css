.Toolbar {
  text-align: left;
  display: flex;
  justify-content: space-between;
  padding: 0.5rem 1rem;
}

/* .SubPage {
  margin: 0.5rem 0 2rem 0.5rem;
  text-align: left;
} */
