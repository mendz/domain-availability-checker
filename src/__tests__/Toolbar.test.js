import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { Toolbar } from '../components/Toolbar/Toolbar';

describe('<Toolbar /> when pathname is "/"', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Toolbar location={{ pathname: '/' }} isChecking={false} />
    );
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should display two buttons go-to-history and ButtonInfo', () => {
    const historyButton = wrapper
      .find('Button')
      .findWhere(comp => comp.prop('name') === 'go-to-history').length;
    const infoButton = wrapper.find('ButtonInfo').length;
    expect(historyButton).toBe(1);
    expect(infoButton).toBe(1);
  });

  it('the buttons should be disabled when the user checking domains', () => {
    wrapper.setProps({ isChecking: true });
    // console.log(wrapper.debug());
    const historyIsDisabled = wrapper
      .find('Button')
      .findWhere(comp => comp.prop('name') === 'go-to-history')
      .props().disabled;
    const infoIsDisabled = wrapper.find('ButtonInfo').props().disabled;
    expect(historyIsDisabled).toBeTruthy();
    expect(infoIsDisabled).toBeTruthy();
  });
});

describe('<Toolbar /> when pathname is not "/"', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Toolbar location={{ pathname: '/history' }} isChecking={false} />
    );
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should display one button to go back to home', () => {
    expect(wrapper.find('Button').html()).toContain('Back Home');
  });
});
