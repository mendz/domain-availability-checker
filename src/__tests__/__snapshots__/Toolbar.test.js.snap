// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> when pathname is "/" matches the snapshot 1`] = `
<div
  className="Toolbar"
>
  <Button
    disabled={false}
    name="go-to-history"
    onClick={[Function]}
  >
    History
  </Button>
  <div
    className="LeftButtons"
  >
    <Connect(AuthModal)
      disabled={false}
    />
    <ButtonInfo
      clicked={[Function]}
      disabled={false}
    />
    <Modal
      closed={[Function]}
      show={false}
    >
      <Info />
    </Modal>
  </div>
</div>
`;

exports[`<Toolbar /> when pathname is not "/" matches the snapshot 1`] = `
<div
  className="Toolbar"
>
  <Button
    onClick={[Function]}
  >
    Back Home
  </Button>
</div>
`;
